realm/realm-library/src/main/cpp/jni_util/hack.cpp

```diff
@@ -72,6 +67,7 @@ static void check_memmove()
     if (ptr != array + 1) {
         Log::e("memmove is broken on this device. switch to the builtin implementation.");
         s_wrap_memmove_ptr = &hacked_memmove;
+        s_wrap_memcpy_ptr =  &hacked_memmove;
```

Comment:
yes, the only difference between `memcpy` and `memmove` is that the `memmove` guarantees to work as expected when the source and destination are overlapping.